demos/node-classification/cluster-gcn/cluster-gcn-node-classification.ipynb

```diff
@@ -198,7 +198,7 @@
        " 'Theory'}"
```

Comment:
<p>This could be made significantly faster by using the sparse adjacency matrix, converting to https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.lil_matrix.html#scipy.sparse.lil_matrix and reading the <code>rows</code> property:</p><p> </p><pre class="ql-syntax" spellcheck="false">lil_adj = G.to_adjacency_matrix().tolil()
adjlist = list(lil_adj.rows)
</pre><p>or</p><pre class="ql-syntax" spellcheck="false">lil_adj = G.to_adjacency_matrix().tolil()
adjlist = [tuple(neighbours) for neighbours in lil_adj.rows]
</pre><p> </p><p>This seems to give the exact same result, except the LIL form results in sorted lists, unlike the NetworkX form:</p><p> </p><pre class="ql-syntax" spellcheck="false">import stellargraph as sg
import networkx as nx

G, _ = sg.datasets.Cora().load()

g_nx = nx.convert_node_labels_to_integers(G.to_networkx())
via_nx = [sorted(neighs) for neighs in nx.to_dict_of_lists(g_nx).values()]

via_lil = list(G.to_adjacency_matrix().tolil().rows)

via_nx == via_lil # True
</pre><p>For my claim of speed:</p><pre class="ql-syntax" spellcheck="false">%timeit -n 10 list(G.to_adjacency_matrix().tolil().rows)
# 5.16 ms ± 505 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit -n 10 list(nx.to_dict_of_lists(nx.convert_node_labels_to_integers(G.to_networkx())).values())
# 88.6 ms ± 1.59 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
</pre><p>(Using a <code>tuple(...)</code>  list comprehension instead of <code>list(...)</code>  doesn't make much difference)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1260/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Reply:
Ok, I couldn't figure out how to get the adjacency list representation directly from the `StellarGraph` object so I went via networkx because this is how `metis` handles the input as well. I updated to avoid the conversion to `NetworkX`. 

Maybe adding a method `StellarGraph.to_adjacency_list()` to wrap `G.to_adjacency_matrix().tolil()` would be helpful?